Basic Mob Flesh Worm

Jacquerel opened the pull request:
## About The Pull Request

Fixes #68614

Converts the Flesh Worm (Armsy) into a Basic Mob.
Most of its behaviour has been moved into a component which we can use to make arbitrary mobs into linked lists of mobs.
To accomplish this I added a signal which is sent when you call any `adjustXLoss` proc, let me know if my implementation is "calling the same signal from several places" by a backdoor, I wanted to avoid registering to 6 signals but I'll change it if I must.

While I was here I killed the unused "lesser" variant because we stopped using it. Resultingly, Ascended Armsy doesn't need to distinguish itself by inflating the sprite, so it doesn't. This means that now flesh worms are using their sprites as intended to be displayed, but if people really miss all of its segments being poorly scaled by the byond engine then I guess I can restore it.

## Why It's Good For The Game

![dreamseeker_p8vOpZGXII](https://github.com/tgstation/tgstation/assets/7483112/3389d3a9-16cd-4e1e-938e-dfa18d0da0af)

## Changelog

:cl:
refactor: Flesh Worms are now basic mobs. Please report any unexpected behaviour.
sprite: Flesh Worms are a little bit slimmer.
/:cl:

Jacquerel committed:
Let's get cracking

Jacquerel committed:
Sorted the basics

Jacquerel committed:
If anything drives me to madness it is probably going to be this component

Jacquerel committed:
we're so fucking back

Jacquerel committed:
Make it easier to do this with VV

Jacquerel committed:
Ok time for the dumb part

Jacquerel committed:
Damage passing
Debug proc to create the Ian centipede

Jacquerel committed:
Unit test
Use leash to catch up

Jacquerel committed:
Merge branch 'master' into worms-good

# Conflicts:
#	code/modules/mob/living/basic/health_adjustment.dm
#	code/modules/mob/living/carbon/damage_procs.dm
#	code/modules/mob/living/damage_procs.dm
#	code/modules/mob/living/simple_animal/damage_procs.dm
#	code/modules/surgery/bodyparts/_bodyparts.dm
#	code/modules/unit_tests/_unit_tests.dm

Sealed101 commented:
closes #68614?

Jacquerel commented:
> closes #68614?

![dreamseeker_KTk7o7dpYG](https://github.com/tgstation/tgstation/assets/7483112/37653d9e-350a-404e-a50e-0f602649cdc7)
Looks a bit weird but seems fixed.

Jacquerel committed:
Make a global list of arm zones

Jacquerel committed:
Missed one

Jacquerel committed:
Restore lost biotypes check, thanks unit tests

Jacquerel committed:
Another win for linters

Jacquerel committed:
Merge branch 'master' into worms-good

san7890 commented on code/modules/admin/verbs/debug.dm:
```diff
@@ -948,6 +948,51 @@
+		return
+	var/mob/living/head = holder.marked_datum
+
+	var/attempted_target_path = input(
```
making worms out of people isn't an admin-critical tool (sadly), make this `tgui_input` instead. you use `tgui_input_number` later on so lets be consistent

san7890 commented on code/modules/admin/verbs/debug.dm:
```diff
@@ -948,6 +948,51 @@
+	set category = "Debug"
+	set name = "Create Mob Worm"
+	set desc = "Attached a linked list of mobs to a marked mob"
+	if (!check_rights(R_DEBUG))
```
`R_FUN` instead imo (or both) (people with `R_DEBUG` can give themselves fun but shhh)

san7890 commented on code/modules/antagonists/heretic/structures/knock_final.dm:
```diff
@@ -21,8 +21,7 @@
 		/mob/living/basic/heretic_summon/star_gazer,
-		/mob/living/simple_animal/hostile/heretic_summon/armsy,
-		/mob/living/simple_animal/hostile/heretic_summon/armsy/prime,
+		/mob/living/basic/heretic_summon/armsy,
```
alphabetize

san7890 commented on code/datums/components/mob_chain.dm:
```diff
@@ -0,0 +1,205 @@
+ * Component allowing you to create a linked list of mobs.
+ * These mobs will follow each other and attack as one, as well as sharing damage taken.
+ */
+/datum/component/mob_chain
```
human centipede simulator just in time for halloween (no change)

san7890 reviewed (changes requested):
nits iirc

Jacquerel committed:
Streamlining admin human centipede experience
Allow chain mobs to pass clicks to floor turfs in case they have a ranged attack

Jacquerel committed:
Removes redundant typecast

LemonInTheDark commented:
> I added a signal which is sent when you call any `adjustXLoss` proc, let me know if my implementation is "calling the same signal from several places" by a backdoor, I wanted to avoid registering to 6 signals but I'll change it if I must.

I think that'd be a better pattern TBH. You aren't the only one who might want to access this info, and they might want it split up by type. Plus you could just feed them all into one proc, so it's just a RegisterSignals and not RegisterSignal

Jacquerel commented:
I'll go back and do it

LemonInTheDark commented:
I appreciate it. don't have to if you don't wanna. I won't hold it against you I just saw this on the bus and was like "wait this is bad", and wanted to say somethin when I got to computer.
